packages/powersync_core/lib/src/database/powersync_db_mixin.dart

```diff
@@ -269,17 +271,30 @@ mixin PowerSyncDatabaseMixin implements SqliteConnection {
   ///
   /// The connection is automatically re-opened if it fails for any reason.
   ///
+  /// To set sync parameters used in your sync rules (if any), use
+  /// [SyncOptions.params]. [SyncOptions] can also be used to tune the behavior
+  /// of the sync client, see that class for more information.
+  ///
   /// Status changes are reported on [statusStream].
   Future<void> connect({
     required PowerSyncBackendConnector connector,
-    Duration crudThrottleTime = const Duration(milliseconds: 10),
-    Map<String, dynamic>? params,
+    SyncOptions? options,
+    @Deprecated('Use SyncOptions.crudThrottleTime instead')
+    Duration? crudThrottleTime,
+    @Deprecated('Use SyncOptions.params instead') Map<String, dynamic>? params,
```

Comment:
I wonder if we should keep this as a separate direct argument here?

I imagine in many applications the params could change dynamically, while the other options would be fairly constant. So to me it makes sense to group all other options (retryDelay, crudThrottleTime, and other future options), but keeping the params separate. We could still group it in the resolved options, to keep the implementation simpler.